src/Microsoft.OData.Core/UriParser/Binders/RangeVariableBinder.cs

```diff
@@ -24,6 +25,16 @@ internal static SingleValueNode BindRangeVariableToken(RangeVariableToken rangeV
         {
             ExceptionUtils.CheckArgumentNotNull(rangeVariableToken, "rangeVariableToken");
 
+            // If $this is the rangeVariableToken Name, we push a rangeVariable with the same TypeReference as the ImplicitRangevariable
+            // to the state.RangeVariables stack.
+            if (rangeVariableToken.Name == ExpressionConstants.This)
+            {
+                // $this range variable will always have the same TypeReference as the ImplicitRangevariable.
+                IEdmTypeReference thisRangeVariableTypeReference = state.ImplicitRangeVariable.TypeReference;

```

Comment:
What does the ImplicitRangeVariable refer to in a query? Is it the same as `$it`?

Reply:
@habbes There is a subtle difference between `implicit range variables` and `explicit range variables`.
In this example, when you parse https://serviceUri.com/Orders?$expand=Items($filter=Product eq 'unknown'), the filter clause has an implicit range variable $it which reference the current expanded entity `Items`. The implicit range variable $it is not defined in the query string.

In the scenario https://serviceUri.com/Orders?$expand=Items($filter=$it/Product eq 'unknown'), where Product is a property in `Order` entity type. The filter clause has an explicit range variable $it that reference the resource from the path (Orders).

Logically, since the ImplicitRangeVariable references the expanded/selected entity, it behaves the same as the $this range variable.